knime_extension/src/nodes/spatialtool.py

```diff
@@ -1152,3 +1157,142 @@ def HaversineDist(x1, y1, x2, y2):
             axis=1,
         )
         return knext.Table.from_pandas(df)
+
+
+############################################
+# Create Voronoi
+############################################
+
+
+@knext.node(
+    name="Voronoi (Thiessen) Polygon",
+    node_type=knext.NodeType.MANIPULATOR,
+    icon_path=__NODE_ICON_PATH + "Voronoi.png",
+    category=__category,
+    after="",
+)
+@knext.input_table(
+    name="Input Point Table",
+    description="Input point data of Voronoi (Thiessen) Polygon",
+)
+@knext.input_table(
+    name="Input Boundary Table",
+    description="Input boundary data for Voronoi (Thiessen) Polygon",
+)
+@knext.output_table(
+    name="Voronoi (Thiessen) Polygon",
+    description="Output table of Voronoi (Thiessen) Polygon",
+)
+class CreateVoronoi:
+    """Create Voronoi (Thiessen) Polygon.
+    This node creates [Voronoi (Thiessen) polygons](https://en.wikipedia.org/wiki/Voronoi_diagram) from the input
+    point data according to the reference boundary. The input data for the boundary reference should be a
+    Polygon or MultiPolygon. The buffer distance (in kilometers) is used to create virtual points to generate dummy
+    points and control the output of Voronoi polygons. If the final output polygon is smaller than the reference
+    boundary, users may consider increasing the buffer distance.
+    """
+
+    point_geo_col = knext.ColumnParameter(
+        "Point geometry column",
+        "Select the geometry column with the points for the Voronoi (Thiessen) polygons.",
+        # Allow only GeoValue compatible columns
+        port_index=0,
+        column_filter=knut.is_geo_point,
+        include_row_key=False,
+        include_none_column=False,
+    )
+
+    boundary_geo_col = knext.ColumnParameter(
+        "Boundary geometry column",
+        "Select the geometry column with the boundary polygon for the Voronoi (Thiessen) polygons.",
+        # Allow only GeoValue compatible columns
+        port_index=1,
+        column_filter=knut.is_geo,
+        include_row_key=False,
+        include_none_column=False,
+    )
+
+    control_buffer = knext.IntParameter(
+        "Buffered distance in kilometers for the Voronoi polygon boundaries.",
+        "The buffer distance for the bounding box of the boundary geometry column.",
```

Comment:
Can you please explain the usage of the distance a bit better.

Reply:
If the buffer distance is too small, the resulting polygons may not fill the entire boundary, and there may be gaps or missing parts within the boundary. This is because the buffer distance determines how far away from the boundary the Voronoi polygons will be generated, and if the distance is too small, some of the polygons may not intersect with the boundary, or may only partially intersect with it.
![image](https://user-images.githubusercontent.com/39640889/222823998-167aa083-055f-4cab-ac2d-b2226f161ca3.png)